site/app/entities/forum/Thread.php

```diff
@@ -157,6 +188,18 @@ public function getPosts(): Collection {
     public function getCategories(): Collection {
         return $this->categories;
     }
+
+    /**
+     * @param Category[]|Collection<Category> $categories
+     * @return void
+     */
+    public function setCategories(array|Collection $categories): void {
+        $new_categories = new ArrayCollection($categories);
+        if (!Category::areCollectionsEqual($this->categories, $new_categories)) {
```

Comment:
Rather than having to implement a custom collection equality. Why not just call `toArray` on them and compare those?